Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1056,33 +1063,104 @@ void PcapLiveDevice::setDefaultGateway()
 			PCPP_LOG_ERROR("Error retrieving default gateway address: " << e.what());
 		}
 	}
-#elif defined(__APPLE__) || defined(__FreeBSD__)
-	std::string command = "netstat -nr | grep default | grep " + m_Name;
-	std::string ifaceInfo = executeShellCommand(command);
-	if (ifaceInfo == "")
-	{
-		PCPP_LOG_DEBUG("Error retrieving default gateway address: couldn't get netstat output");
+#elif defined(__APPLE__)
+
+	//route message struct for communication in APPLE device
+	struct BSDRoutingMessage{
+		struct	rt_msghdr header;
+		char	messageSpace[512];
+	};
+
+	struct BSDRoutingMessage routingMessage;
+	char* spacePtr = routingMessage.messageSpace;
```

Comment:
Please notice the pre-commit failure: https://github.com/seladb/PcapPlusPlus/actions/runs/9585549253/job/26431847597?pr=1379

no need to initialize `spacePtr` here because it's also initialized in line 1104

Reply:
Removed the unused initialization .